scene/animation/animation_node_state_machine.cpp

```diff
@@ -195,6 +195,12 @@ AnimationNodeStateMachineTransition::AnimationNodeStateMachineTransition() {
 
 void AnimationNodeStateMachinePlayback::_set_current(AnimationNodeStateMachine *p_state_machine, const StringName &p_state) {
 	current = p_state;
+	if (is_grouped) {
+		_get_parent_playback(p_state_machine->get_animation_tree())->emit_signal(SNAME("current_changed"), p_state, fading_from);
+	} else {
+		emit_signal(SNAME("current_changed"), p_state, fading_from);
+	}
+
```

Comment:
The `fading_from` changes are made outside of `_set_current()`, so all of them should be made to utilize a function like `_set_fading_from()`, and the signal should fire there separate from `current_changed`.